### ISSUE TITLE
Add `rootof(m::Module)` alongside `pathof(m::Module)` for conveniently getting to package root?

### ISSUE THREAD

**Issue by ianshmean** [JuliaLang/julia#31638]
The package root is often useful for accessing any included examples, which are most commonly stored along side `/src` (one directory up from the result of `Base.pathof(m::Module)`) so would adding the following make sense for convenience? It assumes that `m.jl` is located in `src`, one level away from package root, which I believe is a safe assumption?
```julia
""" 
 rootof(m::Module) 
  
 Return the root directory of `m.jl` file that was used to `import` module `m`, 
 or `nothing` if `m` was not imported from a package. 
 """ 
function rootof(m::Module)
    path = pathof(m::Module)
    path == nothing && return nothing
    return joinpath(splitpath(path)[1:end-2]...)
end
```
Alongside:
https://github.com/JuliaLang/julia/blob/5e56243c3ba0dcacf947958520c9f808ce4e9f1d/base/loading.jl#L265-L278

**Comment by stevengj** [comment]
Since any path in the module that you want can be obtained by `joinpath` on `dirname(pathof(m))`, I'm don't feel like we need another function here.

**Comment by ianshmean** [comment]
But `dirname(pathof(m))` gives you the `src` directory, so you'd have to move up one first. 

Currently:
- `joinpath(dirname(dirname(pathof(m))),"examples/example.jl")`
or
- `joinpath(dirname(pathof(m)),"..","examples/example.jl")`

I'm suggesting:
- `joinpath(rootof(m),"examples/example.jl")`

Just a suggestion. I don't feel strongly

**Comment by stevengj** [comment]
It doesn’t seem worth it to add a function just to avoid typing three characters (`../`)

**Comment by ianshmean** [comment]
Fair enough. I'm going to close this issue/suggestion.

But for the sake of completeness if anyone comes searching.. one of the benefits of the proposed `rootpath(m)` vs the norm you're suggesting @stevengj  is that the path is the actual path, rather than one with an enclosed breadcrumb to traverse `src`. I've added examples above to summarize.

**Comment by goretkin** [comment]
I wonder if this is getting at the difference between a module and a package. The module really is at the `src` directory. But `pathof` only works for modules that result from imported packages, so the distinction here feels blurred.

**Comment by ianshmean** [comment]
@goretkin it's a good point. See what you think of the implementation in #33128 
```julia
"""
    rootof(m::Module)

 Return the root directory of the package that imported module `m`,
 or `nothing` if `m` was not imported from a package.
 """
function rootof(m::Module)
    rootmodule = Base.moduleroot(m)
    path = pathof(rootmodule)
    path === nothing && return nothing
    return joinpath(splitpath(path)[1:end-2]...)
end
```

**Comment by vtjnash** [comment]
Implemented in #33128
